### DISCUSSION TITLE
How to auto resize next/Images in Nextjs 13

### DISCUSSION THREAD

**Discussion by hbatalhaStch** [vercel/next.js#50902] [Help] [answered]
### Summary

Using `next/legacy/image`, I can set a size on an image using the width and height attributes, it will resize correctly, and it will downsize when the screen gets smaller. Makes a lot of sense and works like a charm.

Now with new `next/image` in Nextjs 13 that behavior is gone.
Is there a way to get it back? 
What was the reason for change? Performance?

**Comment by SamKomesarook** [comment, marked as answer]
You can use the [fill prop](https://nextjs.org/docs/app/api-reference/components/image#fill) to make it fill the nearest parent with position "relative", "fixed", or "absolute".

I have found it helpful to set the objectFit of the image to "fill" or "cover", and it should resize based on the parent.

**Reply by hbatalhaStch** [reply]
I tried that but it doesn't work

```ts
import Image from "next/image"

export default function Logo() {
    return (
        <div
            style={{
                height: 45,
                width: 140,
                position: 'relative'
            }}
        >
            <Image
                priority={true}
                src="/images/logo.png"
                alt="Picture of the logo"
                fill
            />
        </div>
    )
}
```

I only got it to change size when I set the parent `width` to a percentage value which didn't look very good

**Reply by SamKomesarook** [reply]
You're setting the parent to a specific height and width, which means it won't be affected by a resize. The parent will have to be resizeable, i.e. a percent, a flex child, etc...

Did you try setting the Image objectFit to 'cover' or 'fill'?

**Reply by hbatalhaStch** [reply]
objectFit is deprecated.

> The parent will have to be resizeable, i.e. a percent, a flex child, etc...

That's what I said, using percent didn't look good.

**Reply by SamKomesarook** [reply]
No, it's [not depreciated](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit#browser_compatibility). I am talking about setting it inside the style prop, not as it's own prop.

In order for it to be responsive to the screen size, the parent will need to be set to a percentage one way or another (percentage, flex child, etc...)

**Reply by hbatalhaStch** [reply]
> No, it's [not depreciated](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit#browser_compatibility). I am talking about setting it inside the style prop, not as it's own prop.

Oh I thought you were talking about the next/image `objectFit` props. I set it to `contain` and it got me what I wanted. 

`next/legacy/image` is still better though ( wish they kept the auto resizing in the new image) but I think I just need to keep tweaking it.

Thank you very much.
